python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -39,8 +39,58 @@
 
 ##################### The I/O functions ####################
 
+def read_data_json(data_file, data_fields):
+    """ Read data from a JSON file.
+
+    Each row of the JSON file represents a data record. The function tries
+    to extract a set of values from the data record.
+
+    Parameters
+    ----------
+    data_file : str
+        The JSON file that contains the data.
+    data_fields : list of str
+        The data fields that we will read data from a data record.
+
+    Returns
+    -------
+    dict : map from data name to data
+    """
+    with open(data_file, 'r', encoding="utf8") as json_file:
+        data_records = []
+        for line in json_file.readlines():
+            record = json.loads(line)
+            data_records.append(record)
+
+    data = {key: [] for key in data_fields}
+    for record in data_records:
+        for key in data_fields:
+            data[key].append(record[key])
```

Comment:
How do we distinguish labels and features?

Reply:
for read/write functions, we don't distinguish them.